src/gamma.jl

```diff
@@ -643,12 +643,18 @@ See also [`logabsgamma`](@ref) for real `x`.
 """
 loggamma(x::Number) = _loggamma(float(x))
 
-function _loggamma(x::Union{Real,BigFloat})
+function _loggamma(x::Real)
     (y, s) = logabsgamma(x)
     s < 0 && throw(DomainError(x, "`gamma(x)` must be non-negative"))
     return y
 end
 
+function _loggamma(x::BigFloat)
+    y = BigFloat()
+    ccall((:mpfr_lngamma, :libmpfr), Cint, (Ref{BigFloat}, Ref{BigFloat}, MPFRRoundingMode), y, x, ROUNDING_MODE[])
+    return y
```

Comment:
Seems like you need a check like:
```jl
signbit(x) && return isinteger(x) ? BigFloat(Inf) : throw(DomainError(x, "`gamma(x)` must be non-negative"))
```

Reply:
Except that `isodd(::BigFloat)` requires Julia 1.7.

Reply:
Alternatively, you can check:
```jl
isnan(y) && !isnan(x) && throw(DomainError(x, "`gamma(x)` must be non-negative"))
```
before the `return y` statement.

Reply:
Yes, I just wanted to suggest the NaN checks since `mpfr_lngamma` already handles negative integers and returns NaN for other negative values.

Reply:
I added a check with `isnan`.